### PR TITLE
Changes to SG vendor add the ability to buy more sidearm ammo type and a smartgun battery.

### PR DESCRIPTION
**Pull request by Huffie56**
# About the pull request

# Explain why it's good for the game
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
balance: adding M44 Marksman Speed Loader and SU-6 Smartpistol Magazine in SG vendor for 10 point each.
balance: adding the ability for to buy smartgun DV9 battery for 15 points in SG vendor.
/:cl:

### CODE REVIEW

**Comment by Huffie56** [comment]
need to still add the option to buy extra SG batteries.

File: code/game/machinery/vending/vendor_types/squad_prep/squad_smartgunner.dm [hunk 1]
```diff
@@ -40,6 +42,7 @@ GLOBAL_LIST_INIT(cm_vending_gear_smartgun, list(
 		list("Whistle", 5, /obj/item/device/whistle, null, VENDOR_ITEM_REGULAR),
 		list("Roller Bed", 5, /obj/item/roller, null, VENDOR_ITEM_REGULAR),
 		list("Fulton Device Stack", 5, /obj/item/stack/fulton, null, VENDOR_ITEM_REGULAR),
+		list("smartgun DV9 battery", 15, /obj/item/smartgun_battery, null, VENDOR_ITEM_REGULAR),
```
Comment (Drulikar) [inline code/game/machinery/vending/vendor_types/squad_prep/squad_smartgunner.dm:45]:
```suggestion
		list("Smartgun DV9 Battery", 15, /obj/item/smartgun_battery, null, VENDOR_ITEM_REGULAR),
```

**Review by Drulikar — CHANGES_REQUESTED** [review]
I am not really keen on the primary goal of this PR: moving the SG drums from prep vendors to the SG vendor. I don't think foxtrot SGs need the additional ammo.

**Comment by Huffie56** [comment]
> I am not really keen on the primary goal of this PR: moving the SG drums from prep vendors to the SG vendor. I don't think foxtrot SGs need the additional ammo.

balance wise if it's better i can move the extra SG drum to req instead of adding it to the SG belt?

**Comment by Drulikar** [comment]
> balance wise if it's better i can move the extra SG drum to req instead of adding it to the SG belt.

I'd prefer it left as is.

File: code/game/objects/items/storage/belt.dm [hunk 2]
```diff
@@ -569,6 +569,13 @@
 	new /obj/item/ammo_magazine/smartgun(src)
 	new /obj/item/ammo_magazine/smartgun(src)
 
+/obj/item/storage/belt/marine/smartgunner/uscm
+
+/obj/item/storage/belt/marine/smartgunner/uscm/filled/fill_preset_inventory()
+	new /obj/item/ammo_magazine/smartgun(src)
+	new /obj/item/ammo_magazine/smartgun(src)
+	new /obj/item/ammo_magazine/smartgun(src)
+
```
[the remark this answers is not in the document]
Comment (Huffie56) [reply code/game/objects/items/storage/belt.dm:578]: ok

File: code/game/machinery/vending/vendor_types/squad_prep/squad_smartgunner.dm [hunk 3]
```diff
@@ -81,8 +84,8 @@ GLOBAL_LIST_INIT(cm_vending_clothing_smartgun, list(
 		list("Map", 0, /obj/item/map/current_map, MARINE_CAN_BUY_KIT, VENDOR_ITEM_MANDATORY),
 
 		list("BELT", 0, null, null, null),
-		list("M802 Smartgunner Sidearm Belt", 0, /obj/item/storage/belt/gun/smartgunner/full, MARINE_CAN_BUY_BELT, VENDOR_ITEM_MANDATORY),
-		list("M280 Smartgunner Drum Belt", 0, /obj/item/storage/belt/marine/smartgunner, MARINE_CAN_BUY_BELT, VENDOR_ITEM_MANDATORY),
+		list("M802 Smartgunner Sidearm Belt", 0, /obj/item/storage/belt/gun/smartgunner/uscm/filled, MARINE_CAN_BUY_BELT, VENDOR_ITEM_MANDATORY),
+		list("M280 Smartgunner Drum Belt", 0, /obj/item/storage/belt/marine/smartgunner/uscm/filled, MARINE_CAN_BUY_BELT, VENDOR_ITEM_MANDATORY),
```
Comment (Huffie56) [inline code/game/machinery/vending/vendor_types/squad_prep/squad_smartgunner.dm:89]: ok

File: code/game/machinery/vending/vendor_types/squad_prep/squad_smartgunner.dm [hunk 4]
```diff
@@ -81,8 +84,8 @@ GLOBAL_LIST_INIT(cm_vending_clothing_smartgun, list(
...
```
[the remark this answers is not in the document]
Comment (Huffie56) [reply code/game/machinery/vending/vendor_types/squad_prep/squad_smartgunner.dm:88]: ok

File: code/game/objects/items/storage/belt.dm [hunk 5]
```diff
@@ -1658,6 +1665,15 @@
...
 
+/obj/item/storage/belt/gun/smartgunner/uscm
+
+/obj/item/storage/belt/gun/smartgunner/uscm/filled/fill_preset_inventory()
+	handle_item_insertion(new /obj/item/weapon/gun/pistol/m4a3())
+	new /obj/item/ammo_magazine/pistol/hp(src)
+	new /obj/item/ammo_magazine/smartgun(src)
+	new /obj/item/ammo_magazine/smartgun(src)
+	new /obj/item/ammo_magazine/smartgun(src)
+
```
[the remark this answers is not in the document]
Comment (Huffie56) [reply code/game/objects/items/storage/belt.dm:1676]: ok

**Review by Drulikar — APPROVED** [review]

**Pull request merged**